rfcs/0004-fulltext-search.md

```diff
@@ -0,0 +1,236 @@
+system complexity and feature implementation time I propose starting with the 
+fulltext search features built in to PostgreSQL.
+
+A FullText search field will get its own column in the Entity table just 
+like any other field; however, the data stored will be the result of the 
+lexical, morphological analysis of text documents performed on the fields 
+included in the index. The fulltext search field will be created using 
+Postgres ts_vector function and will be indexed using a GIN index. 
+The GraphQL query interface will expose PostgreSQL's to_tsquery(), 
+ts_rank(), and ts_rank_cd() functions in a pretty direct manner 
+(see the GraphQL Query Interface section above for details) for querying 
+the indexed fulltext search data.  
+ 
+
+Limitations of Postgres fulltext search indexes and search algorithms: 
+  - only certain languages available (expandable with plugins like [PGroonga](https://pgroonga.github.io/)), 
+  - select algorithms are available, and 
+  - limited to PostgreSQL storage. 
+
+Alternative technologies (not in any particular order): 
+  - Written in Rust:
+    - [Tantivy](https://github.com/tantivy-search/tantivy)
+    - [Toshi](https://github.com/toshi-search/Toshi)
+    - [Sonic](https://github.com/valeriansaliou/sonic)
+    - [MeiliSearch](https://github.com/meilisearch/MeiliSearch)
+    - [Bayard](https://github.com/bayard-search/bayard)
+
+## Compatibility
+
+This proposal does not change any existing interfaces, so no migrations 
+will be necessary for existing subgraph deployments. 
+
+## Drawbacks and Risks
+
+There is little risk in implementing this feature.  The proposed solution 
+uses native Postgres fulltext features and there is a nonzero probability 
+this choice results in slower than optimal write and read times; 
+however the tradeoff in implementation time/complexity and the existence 
+of production use case testimonials tempers my apprehension here. 
+
+Are there any desired "fulltext search like" capabilities that I've missed in this proposal?   
+
+## Alternatives
+
+An alternative design for the feature would allow more flexibility for 
+Graph Node operators in their index implementation and create a marketplace 
+for indexes. This design moved the definition of custom indexes out of the 
+subgraph definition; the subgraph would be deployed without them and they 
+could be added later using a new Graph Explorer interface (in Hosted-Service context) 
+or a JSON-RPC request directly to a Graph Node. One of the benefits of this 
+design is that the definition of uniqueness for a subgraph does not include 
+the custom indexes, so a new subgraph deployment and corresponding syncing 
+work does not have to be added in order to create or update an index. 
+However, it also introduces significant added complexity: will a separate 
+query marketplace and discovery registry then be required for finding nodes 
+with your needed subgraph-index combination.  
```

Comment:
I feel we should nonetheless include as little algorithmic details in what is included in the subgraph manifest (and related files). Primarily, what should be included is anything that is required to generate an API that includes fulltext search.

An approach different from declaring _indexes_ could be to declare special fulltext query fields:
```
type _Schema_
  @fulltextAPI(name: "search", entities: [ { name: "ExampleItem", fields: [ ... ] } ])
```
or similar.